packages/sds/src/message_channel/persistent_history.ts

```diff
@@ -0,0 +1,177 @@
+import { bytesToHex, hexToBytes } from "@noble/hashes/utils";
+
+import { MemLocalHistory } from "./mem_local_history.js";
+import { ChannelId, ContentMessage, HistoryEntry } from "./message.js";
+
+export interface HistoryStorage {
+  getItem(key: string): string | null;
+  setItem(key: string, value: string): void;
+  removeItem(key: string): void;
+}
+
+export interface PersistentHistoryOptions {
+  channelId: ChannelId;
+  storage?: HistoryStorage;
+  storageKey?: string;
+}
+
+type StoredHistoryEntry = {
+  messageId: string;
+  retrievalHint?: string;
+};
+
+type StoredContentMessage = {
+  messageId: string;
+  channelId: string;
+  senderId: string;
+  lamportTimestamp: string;
+  causalHistory: StoredHistoryEntry[];
+  bloomFilter?: string;
+  content: string;
+  retrievalHint?: string;
+};
+
+const HISTORY_STORAGE_PREFIX = "waku:sds:history:";
+
+/**
+ * Persists the SDS local history in a browser/localStorage compatible backend.
+ *
+ * If no storage backend is available, this behaves like {@link MemLocalHistory}.
+ */
+export class PersistentHistory extends MemLocalHistory {
+  private readonly storage?: HistoryStorage;
+  private readonly storageKey: string;
+
+  public constructor(options: PersistentHistoryOptions) {
+    super();
+    this.storage = options.storage ?? getDefaultHistoryStorage();
+    this.storageKey =
+      options.storageKey ?? `${HISTORY_STORAGE_PREFIX}${options.channelId}`;
```

Comment:
I wouldnt let the dev should the whole storage key to avoid footgun. Maybe just expose a prefix to apply.
I would also make it mandatory to avoid clashes